src/main/java/utils/Utils.java

```diff
@@ -32,24 +32,25 @@ public static void writeToFileOrStdout(String file, Object value) {
       System.out.println(value);
     } else {
       try {
-        new BufferedWriter(new FileWriter(file)).write(value.toString() + '\n');
+        BufferedWriter writer = new BufferedWriter(new FileWriter(file));
+        writer.write(value.toString() + "\n");
+        writer.flush();
       } catch (IOException e) {
         throw new RuntimeException(e);
       }
     }
   }
 
   public static String readFromFileOrStdin(String file) {
-    if (file == null) {
-      return new Scanner(inReader).next("\\A");
-    } else {
+    if (file != null) {
       try {
-        return new Scanner(new BufferedReader(new FileReader(file)))
-            .next("\\A");
-      } catch (IOException e) {
-        throw new RuntimeException(e);
+        return new Scanner(new FileReader(file)).useDelimiter("\\A")
+                .next();
+      } catch (FileNotFoundException e) {
+        e.printStackTrace();
```

Comment:
Could you change this to throw a new runtime Exception? We don't want this to silently fail and then block on user input.